src/User/UserContainer.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect } from "react";
+import { Switch, Route, useParams } from "react-router-dom";
+import ProfilePage from "./ProfilePage";
+import { getUserById } from "../api.js";
+
+const UserContainer = () => {
+  const [user, setUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const { id } = useParams();
+
+  const fetchData = async userId => {
+    setIsLoading(true);
+
+    try {
+      const user = await getUserById(userId);
+      setUser(user);
+    } catch (error) {
+      setError(true);
+    }
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    if (id) {
+      fetchData(id);
+    }
+  }, [id]);
+
+  return (
+    <Switch>
+      <Route path="/user/:id">
```

Comment:
I'm slightly confused as you have the path to get here as /users but then in here it's changed to /user